youtube_dlc/extractor/vlive.py

```diff
@@ -119,6 +116,46 @@ def _real_extract(self, url):
             PARAMS_RE, webpage, PARAMS_FIELD, default='', flags=re.DOTALL)
         params = self._parse_json(params, working_id, fatal=False)
 
+        return working_id, webpage, params
+
+    def _real_extract(self, url):
+        working_id, webpage, params = self._get_webpage_info(url)
+
+        playlist = try_get(
+            params, lambda x: x['postDetail']['post']['playlist']['data'], list)
+        if not playlist or self._downloader.params.get('noplaylist'):
+            if playlist:
+                self.to_screen(
+                    'Downloading just video %s because of --no-playlist'
+                    % working_id)
+
+            return self._parse_vlive_info(url, working_id, webpage, params)
+        else:
+            playlist_name = try_get(
+                params, lambda x: x['postDetail']['post']['playlist']['name'], compat_str)
+            playlist_id = try_get(
+                params, lambda x: x['postDetail']['post']['playlist']['playlistSeq'], int)
+
+            self.to_screen(
+                'Downloading playlist %s - add --no-playlist to just download video'
```

Comment:
I'm not a fan of defaulting to downloading a detected playlist when given the link for a single video. I think users would likely miss this message to add `--no-playlist` until they realize a playlist is being downloaded. If it is possible to have the default be downloading the single video and switching to downloading a playlist when given an explicit flag that would be better.

Reply:
While I agree with the concern, I think defaulting `--no-playlist` to false goes against consistency of the program.
Even implementation wise, https://github.com/blackjack4494/yt-dlc/blob/master/youtube_dlc/options.py#L331 specifies that `noplaylist` defaults to false, and I can't see a way to only set this true only for vlive.

Also, if `--no-playlist` defaults to false, the opposite case can also happen.
Where a user intends to download a playlist using vidoe link, and only download a single video.
Because the program defaults to downloading all videos within a playlist for other links, the user will think that the program is broken and the program is unable to recognize that the link contains a playlist.

Reply:
Ok, I think because I wasn't that familiar with the `--no-playlist` and `--yes-playlist` flags I thought that normally we avoid downloading playlists unless the user opts in. I think the way it works here should be consistent then.